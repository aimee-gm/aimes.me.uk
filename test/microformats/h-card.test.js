const { expect } = require("chai");

const { readMicroformats } = require("../helpers/fileSystem");

describe("microformats // h-card", () => {
  it("should match parsed snapshot", async () => {
    const parsed = await readMicroformats("/");
    const hCard = parsed.items[0];

    expect(hCard.properties.name).to.eql(["Aimee Gamble-Milner"]);
    expect(hCard).to.matchSnapshot();
  });
});
