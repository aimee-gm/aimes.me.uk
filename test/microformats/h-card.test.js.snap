// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`microformats // h-card should match parsed snapshot 1`] = `
Object {
  "properties": Object {
    "country-name": Array [
      "UK",
    ],
    "honorific-suffix": Array [
      "PhD",
    ],
    "job-title": Array [
      "Senior Software Engineer",
    ],
    "locality": Array [
      "London",
    ],
    "name": Array [
      "Aimee Gamble-Milner",
    ],
    "org": Array [
      Object {
        "properties": Object {
          "name": Array [
            "Bulb",
          ],
          "url": Array [
            "https://bulb.co.uk",
          ],
        },
        "type": Array [
          "h-card",
        ],
        "value": "Bulb",
      },
    ],
    "photo": Array [
      Object {
        "alt": "Aimee Gamble-Milner's profile picture",
        "value": "https://media.aimes.me.uk/images/2020/02/04/southwark.jpg",
      },
    ],
    "uid": Array [
      "https://www.aimes.me.uk",
    ],
    "url": Array [
      "https://www.aimes.me.uk",
    ],
    "x-pronoun-nominative": Array [
      "she",
    ],
    "x-pronoun-oblique": Array [
      "her",
    ],
  },
  "type": Array [
    "h-card",
  ],
}
`;
